### PR TITLE
[plug-in] Do not restart backend plugins when they already started

### PR DESCRIPTION
**Pull request by evidolob**
Should also fix hosted mode for backend plugins.

### CODE REVIEW

File: packages/plugin-ext/src/hosted/node/hosted-plugin.ts [hunk 1]
```diff
@@ -51,11 +53,13 @@ export class HostedPluginSupport {
     }
 
     setClient(client: HostedPluginClient): void {
+        this.client = client;
         this.hostedPluginProcess.setClient(client);
         this.pluginRunners.forEach(runner => runner.setClient(client));
     }
 
     clientClosed(): void {
+        this.isPluginProcessRunning = false;
```
Comment (benoitf) [inline packages/plugin-ext/src/hosted/node/hosted-plugin.ts]: I don't see when clientClosed is called ?

Comment (evidolob) [reply packages/plugin-ext/src/hosted/node/hosted-plugin.ts]: @benoitf [there](https://github.com/theia-ide/theia/blob/8ca8c80044001c20c20a36384de4e3202f541388/packages/plugin-ext/src/hosted/node/plugin-service.ts#L50)

Comment (benoitf) [reply packages/plugin-ext/src/hosted/node/hosted-plugin.ts]: yes but https://github.com/theia-ide/theia/blob/master/packages/plugin-ext/src/hosted/node/plugin-ext-hosted-backend-module.ts#L50

Comment (benoitf) [reply packages/plugin-ext/src/hosted/node/hosted-plugin.ts]: what is happening if you refresh the browser ?

Comment (evidolob) [reply packages/plugin-ext/src/hosted/node/hosted-plugin.ts]: When set new client we restart plugin node: https://github.com/theia-ide/theia/blob/8ca8c80044001c20c20a36384de4e3202f541388/packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts#L42

Comment (benoitf) [reply packages/plugin-ext/src/hosted/node/hosted-plugin.ts]: ah yes

**Pull request merged**